src/Api/Orbit.cpp

```diff
@@ -30,29 +38,21 @@ void EnqueueApiEvent(Types... args) {
   Event event{pid, tid, timestamp_ns, args...};
   producer.EnqueueIntermediateEvent(event);
 }
-}  // namespace
 
-extern "C" {
+void orbit_api_start_v1(const char* name, orbit_api_color color, uint64_t group_id) {
+  auto return_address = ORBIT_GET_CALLER_PC();
+  EnqueueApiEvent<orbit_api::ApiScopeStart>(name, color, group_id, return_address);
+}
 
 void orbit_api_start(const char* name, orbit_api_color color) {
-  // `__builtin_return_address(0)` will return us the (possibly encoded) return address of the
-  // current function (level "0" refers to this frame, "1" would be the callers return address and
-  // so on).
-  // To decode the return address, we call `__builtin_extract_return_addr`.
-  auto return_address =
-      absl::bit_cast<uint64_t>(__builtin_extract_return_addr(__builtin_return_address(0)));
-  EnqueueApiEvent<orbit_api::ApiScopeStart>(name, color, 0LU /*group_id*/, return_address);
+  auto return_address = ORBIT_GET_CALLER_PC();
+  EnqueueApiEvent<orbit_api::ApiScopeStart>(name, color, kOrbitDefaultGroupId, return_address);
 }
 
 void orbit_api_stop() { EnqueueApiEvent<orbit_api::ApiScopeStop>(); }
 
 void orbit_api_start_async(const char* name, uint64_t id, orbit_api_color color) {
-  // `__builtin_return_address(0)` will return us the (possibly encoded) return address of the
-  // current function (level "0" refers to this frame, "1" would be the callers return address and
-  // so on).
-  // To decode the return address, we call `__builtin_extract_return_addr`.
-  auto return_address =
-      absl::bit_cast<uint64_t>(__builtin_extract_return_addr(__builtin_return_address(0)));
+  auto return_address = ORBIT_GET_CALLER_PC();
```

Comment:
auto -> uint64_t, or use ORBIT_GET_CALLER_PC() directly in the function call maybe?

Reply:
Done.